envoy/stream_info/stream_info.h

```diff
@@ -593,6 +593,17 @@ class StreamInfo {
    * @return Network filter chain name of the downstream connection.
    */
   virtual const std::string& filterChainName() const PURE;
+
+  /**
+   * @param attempt_count, the number of times the request was attempted upstream.
+   */
+  virtual void setAttemptCount(uint32_t attempt_count) PURE;
```

Comment:
Thought for a future day: do we want to split StreamInfo into downstream and upstream? Some fields are only applicable to one or the other; attempt count only makes sense on the downstream side.

Reply:
I had similar thoughts while having a [discussion](https://github.com/envoyproxy/envoy/pull/17512#discussion_r681610376) with @soulxu on a similar PR to this one. 

Created https://github.com/envoyproxy/envoy/issues/17594